File: tags/math/pairs-of-songs-with-total-durations-divisible-by-60/Solution.go
func numPairsDivisibleBy60(time []int) int {
    var res int
    seconds := make([]int, 60)
    for _,v := range time {
        seconds[v % 60]++
    }
    res += seconds[0] * (seconds[0] - 1) / 2
    res += seconds[30] * (seconds[30] - 1) / 2
    i := 1
    j := 59
    for i < j {
        res += seconds[i] * seconds[j]
        i++
        j--
    }
    return res

}

File: tags/math/pairs-of-songs-with-total-durations-divisible-by-60/Solution.java
class Solution {
    public int numPairsDivisibleBy60(int[] time) {
        int res = 0;
        int[] seconds = new int[60];
        for(int t : time){
            seconds[t % 60]++;
        }
        res += (seconds[30]) * (seconds[30] - 1) / 2;
        res += (seconds[0]) * (seconds[0] - 1) / 2;
        int i = 1,j = 59;
        while(i < j){
            res += seconds[i++] * seconds[j--];
        }
        return res;
    }
}